metrics/options/options.go

```diff
@@ -36,14 +38,20 @@ type HeapsterRunOptions struct {
 }
 
 func NewHeapsterRunOptions() *HeapsterRunOptions {
-	return &HeapsterRunOptions{}
+	opt := genericoptions.NewServerRunOptions()
+	return &HeapsterRunOptions{
+		ServerRunOptions: opt,
+	}
 }
 
 func (h *HeapsterRunOptions) AddFlags(fs *pflag.FlagSet) {
+	h.ServerRunOptions.AddUniversalFlags(pflag.CommandLine)
+
 	fs.Var(&h.Sources, "source", "source(s) to watch")
 	fs.Var(&h.Sinks, "sink", "external sink(s) that receive data")
 	fs.DurationVar(&h.MetricResolution, "metric_resolution", 60*time.Second, "The resolution at which heapster will retain metrics.")
-	fs.IntVar(&h.Port, "port", 8082, "port to listen to")
+	// TODO: Revise this flag before Heapster v1.3 and Kubernetes v1.5
+	fs.IntVar(&h.Port, "heapster-port", 8082, "port used by the Heapster-specific APIs")
```

Comment:
Why do we want to expose the heapster specific APIs on a different port?

Changing the meaning of an existing flag doesnt seem right to me.
If I was setting --port=8090 in my config and was expecting to see a heapster specific API at that port, then this change will break me.